controllers/chat/sendSignedMessage.js

```diff
@@ -0,0 +1,79 @@
+const Validator = require('fastest-validator');
+const {StreamChat} = require('stream-chat');
+const {responseError, responseSuccess} = require('../../utils/Responses');
+const {CHANNEL_TYPE} = require('../../helpers/constants');
+
+const v = new Validator();
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const sendSignedMesage = async (req, res) => {
+  const schema = {
+    channelId: 'string|empty:false',
+    message: 'string|empty:false',
+    channelType: {
+      type: 'enum',
+      values: Object.values(CHANNEL_TYPE),
+      empty: false
+    }
+  };
+  const validated = v.validate(req.body, schema);
+  if (validated.length)
+    return res.status(403).json({
+      message: 'Error validation',
+      error: validated
+    });
+
+  const {channelId, message, channelType} = req.body;
+
+  let channelTypeDef;
+
+  switch (channelType) {
+    case CHANNEL_TYPE.CHAT:
+    case CHANNEL_TYPE.ANONYMOUS:
+      channelTypeDef = 'messaging';
+      break;
+
+    case CHANNEL_TYPE.GROUP:
+      channelTypeDef = 'group';
+      break;
+
+    default:
+      return res.status(403).json({
+        message: 'Error validation',
+        error: 'Channel type not found'
+      });
+  }
+
+  const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+  try {
+    await client.connectUser({id: req.userId}, req.token);
+
+    const channel = client.channel(channelTypeDef, channelId);
+
+    const createdChannel = await channel.create();
+
+    if (createdChannel?.channel?.is_channel_blocked)
+      return res.status(403).json(responseError('Channel is blocked'));
+
+    const chat = await channel.sendMessage({
+      user_id: req.userId,
+      text: message
+    });
+
+    await client.disconnectUser();
+    return res.status(200).json(responseSuccess('sent', chat));
+  } catch (error) {
+    await client.disconnectUser();
+    return res.status(error.statusCode ?? error.status ?? 500).json({
+      status: 'error',
+      code: error.statusCode ?? error.status ?? 500,
+      message: error.message
+    });
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for connecting to the Stream Chat API, sending a message, and handling errors is well implemented. However, consider wrapping the `client.connectUser` and `client.disconnectUser` calls in a `finally` block to ensure that the user is always disconnected, even if an error occurs before the disconnect call in the `catch` block.

```diff
-  try {
-    await client.connectUser({id: req.userId}, req.token);
-
-    const channel = client.channel(channelTypeDef, channelId);
-
-    const createdChannel = await channel.create();
-
-    if (createdChannel?.channel?.is_channel_blocked)
-      return res.status(403).json(responseError('Channel is blocked'));
-
-    const chat = await channel.sendMessage({
-      user_id: req.userId,
-      text: message
-    });
-
-    await client.disconnectUser();
-    return res.status(200).json(responseSuccess('sent', chat));
-  } catch (error) {
-    await client.disconnectUser();
-    return res.status(error.statusCode ?? error.status ?? 500).json({
-      status: 'error',
-      code: error.statusCode ?? error.status ?? 500,
-      message: error.message
-    });
-  }
+  try {
+    await client.connectUser({id: req.userId}, req.token);
+
+    const channel = client.channel(channelTypeDef, channelId);
+
+    const createdChannel = await channel.create();
+
+    if (createdChannel?.channel?.is_channel_blocked)
+      return res.status(403).json(responseError('Channel is blocked'));
+
+    const chat = await channel.sendMessage({
+      user_id: req.userId,
+      text: message
+    });
+
+    return res.status(200).json(responseSuccess('sent', chat));
+  } catch (error) {
+    return res.status(error.statusCode ?? error.status ?? 500).json({
+      status: 'error',
+      code: error.statusCode ?? error.status ?? 500,
+      message: error.message
+    });
+  } finally {
+    await client.disconnectUser();
+  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->